### DISCUSSION TITLE
How to enable debug logging for Windows image builds

### DISCUSSION THREAD

**Discussion by hendrik-schaffer** [actions/virtual-environments#10538] [General questions] [answered]
Dear Community,

does anyone know how to enable the debug messages for the Windows images? I am referring to the debug messages in the installer scripts etc, by "Write-Debug". 

I tried adding `$global:DebugPreference = "Continue"` to the Powershell but still, the debug messages during the build are not printed out to the shell. 

```
      - name: Trigger image creation
        run: |
          $global:DebugPreference = "Continue"
          .\images.CI\linux-and-win\build-image.ps1 [...build parameters]
        shell: pwsh

```

Any ideas of what I am missing?

**Comment by Prabhatkumar59** [comment, marked as answer]
Hi @hendrik-schaffer - To enable debug logging in PowerShell scripts for Windows image builds, you need to set the **DebugPreference** to "**Continue**" and ensure that **Write-Debug** messages are actually written to the console. 

**For you, I am providing some steps to troubleshoot and ensure debug logging works**:-

A. Ensure **DebugPreference** is correctly set: The correct setting for enabling debug messages is $DebugPreference = "Continue". Make sure this is set in the correct scope (within the script or globally, as needed).

B. Check script content for **Write-Debug**: Ensure that your PowerShell script (build-image.ps1) is using Write-Debug for logging messages, as opposed to Write-Host or Write-Output.

C. Set **DebugPreference** before the script runs: Instead of setting DebugPreference inside the same run block, set it globally before invoking the script. Here's an updated version:

```
- name: Trigger image creation
  run: |
    $DebugPreference = "Continue"
    .\images.CI\linux-and-win\build-image.ps1 [...build parameters]
  shell: pwsh
```
D. Verify script invocation: You can also Check if the script is executing as expected. If you are redirecting output or running it in an environment where the console output is limited, you might need to adjust how the logs are captured.
